Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+        -> JsonSerializationResult::ResultCode
+    {
+        using ReporterString = AZStd::fixed_string<1024>;
+
+        if (auto importedObject = JsonSerializationUtils::ReadJsonFile(importAbsPath.Native());
+            importedObject.IsSuccess())
+        {
+            JsonApplyPatchSettings applyPatchSettings;
+            applyPatchSettings.m_reporting = issueReporter;
+
+            rapidjson::Value& importedDoc = importedObject.GetValue();
+            if (importDirective.IsObject())
+            {
+                auto patchField = importDirective.FindMember("patch");
+                if (patchField != importDirective.MemberEnd())
+                {
+                    JsonSerialization::ApplyPatch(importedDoc, allocator, patchField->value,
+                        AZ::JsonMergeApproach::JsonMergePatch, applyPatchSettings);
+                }
+            }
+
+            constexpr AZStd::string_view JsonPatchExtension = ".jsonpatch";
+            constexpr AZStd::string_view JsonPatchExtensionForSetreg = ".setregpatch";
+            const auto mergeApproach = importAbsPath.Extension() != JsonPatchExtension
+                && importAbsPath.Extension() != JsonPatchExtensionForSetreg
+                ? AZ::JsonMergeApproach::JsonMergePatch : AZ::JsonMergeApproach::JsonPatch;
+
+            JsonSerializationResult::ResultCode importResult = JsonSerialization::ApplyPatch(target, allocator, importedDoc,
+                mergeApproach, applyPatchSettings);
+
+            if (importResult.GetProcessing() != JsonSerializationResult::Processing::Completed)
+            {
+                return issueReporter(
+                    ReporterString::format(R"(Patching the imported JSON file "%.*s" into the target JSON document has failed.)",
+                        AZ_PATH_ARG(importAbsPath)),
+                    importResult, element.Get());
+            }
+
+            return issueReporter(
+                ReporterString::format(R"(Successfully merged imported JSON file "%.*s into the target JSON document".)",
+                    AZ_PATH_ARG(importAbsPath)),
+                importResult, element.Get());
+        }
+        else
+        {
+            return issueReporter(
+                ReporterString::format(R"(Failed to import file "%.*s". Reason: "%s")",
+                    AZ_PATH_ARG(importAbsPath), importedObject.GetError().c_str()),
+                JsonSerializationResult::ResultCode(JsonSerializationResult::Tasks::Import,
+                    JsonSerializationResult::Outcomes::Catastrophic), element.Get());
+        }
+    }
+
+    JsonSerializationResult::ResultCode JsonImportResolver::ResolveNestedImportsInOrder(rapidjson::Value& target,
+        rapidjson::Document::AllocatorType& allocator, const rapidjson::Value& source,
+        ImportPathStack& importPathStack,
+        JsonImportSettings& settings, StackedString& element, AZ::IO::PathView importPath)
```

Comment:
There are a lot of parameters on these new methods. Just double-checking - do the order of the parameters on these make logical sense from a caller point of view? I'm just asking because the placement of target vs source and importPath vs importPathStack seem a bit weird to me just looking at it, but maybe it makes more sense in the larger context. 

Either way, I don't have any blocking concerns since the parameters are well-named and easy to understand.

Reply:
Well this function in particular is a private function.

The arguments are in a simliar order to the `ResolveNestedImports` function